### DISCUSSION TITLE
Irregular weeks, months, and lunar cycles

### DISCUSSION THREAD

**Discussion by dougseay** [vigoren/foundryvtt-simple-calendar#174] [Q&A]
In a home grown setting I occasionally run, the calendar is basically the same as Earth's Gregorian. But the lunar cycle matches the month, so that the first of the month is always a new moon, and the middle of the month is a full moon, and is either the 14th or 15th of a month. Since months vary from 28-31 days, I made weeks variable of sometimes 6. sometimes 7, sometimes 8 days.

So far Leap Year hasn't come up, so I don't know how I'd handle it. I know I'd be tempted to double February 24th as in the Julian Calendar because I'm a glutton for punishment. But for now I have no need of a Leap Day.

I realize that this is bizarre. Is there any hope in fitting this into simple-calendar?

<big> <table xmlns="http://www.w3.org/1888/xhtml" border="1" data-table-local-id="table-2">
  <tbody>
    <tr>
      <th> Month </th>
      <th> days </th>
      <th> New Moon </th>
      <th> First Quarter </th>
      <th> Full Moon </th>
      <th> Last Quarter </th>
    </tr>
    <tr>
      <td> January </td>
      <td> 31 </td>
      <td> 1st </td>
      <td> 8th </td>
      <td> 15th </td>
      <td> 23rd </td>
    </tr>
    <tr>
      <td> February </td>
      <td> 28 </td>
      <td> 1st </td>
      <td> 7th </td>
      <td> 14th </td>
      <td> 21st </td>
    </tr>
    <tr>
      <td> March </td>
      <td> 31 </td>
      <td> 1st </td>
      <td> 8th </td>
      <td> 15th </td>
      <td> 23rd </td>
    </tr>
    <tr>
      <td> April </td>
      <td> 30 </td>
      <td> 1st </td>
      <td> 7th </td>
      <td> 14th </td>
      <td> 22nd </td>
    </tr>
    <tr>
      <td> May </td>
      <td> 31 </td>
      <td> 1st </td>
      <td> 8th </td>
      <td> 15th </td>
      <td> 23rd </td>
    </tr>
    <tr>
      <td> June </td>
      <td> 30 </td>
      <td> 1st </td>
      <td> 7th </td>
      <td> 14th </td>
      <td> 22nd </td>
    </tr>
    <tr>
      <td> July </td>
      <td> 31 </td>
      <td> 1st </td>
      <td> 8th </td>
      <td> 15th </td>
      <td> 23rd </td>
    </tr>
    <tr>
      <td> August </td>
      <td> 31 </td>
      <td> 1st </td>
      <td> 8th </td>
      <td> 15th </td>
      <td> 23rd </td>
    </tr>
    <tr>
      <td> September </td>
      <td> 30 </td>
      <td> 1st </td>
      <td> 7th </td>
      <td> 14th </td>
      <td> 22nd </td>
    </tr>
    <tr>
      <td> October </td>
      <td> 31 </td>
      <td> 1st </td>
      <td> 8th </td>
      <td> 15th </td>
      <td> 23rd </td>
    </tr>
    <tr>
      <td> November </td>
      <td> 30 </td>
      <td> 1st </td>
      <td> 7th </td>
      <td> 14th </td>
      <td> 22nd </td>
    </tr>
    <tr>
      <td> December </td>
      <td> 31 </td>
      <td> 1st </td>
      <td> 8th </td>
      <td> 15th </td>
      <td> 23rd </td>
    </tr>
  </tbody>
</table> </big>

**Comment by vigoren** [comment]
Interesting calendar!

This is a tricky one. The moon is designed around having a cycle of x days and those days don't have to map directly to specific days of the month. Since our calendar does not have a consistent number of days each month this makes it really hard to get things to map to how you want it.

I can get it "close" but not exactly has you have it, but maybe its a starting point for you to tweak and get closer.

These are the settings I did.

- Load up the predefined Gregorian calendar (to start)
- Under the **Leap Year Settings** tab I change the **Leap Year Rule** to 'None'
- Under the **Moon Settings** tab I made the following changes:
    - **Cycle Length** I changed to 30.41666 (365 / 12)
    - **Cycle Adjust** I changed to 0
    - **Reference Moon Year Reset** I changed this to 'Reset reference year every X years'
    - **Reset Reference Moon Years** I change this to 1 (reset every year)
    - **New Moon Month** Left this as January
    - **New Moon Day** set this to 1

What this does is make January 1st a new moon every year and starts plotting the moon from there. The other phases for January don't quite line up but adjusting the Cycle Length or Cycle Adjust can get them in there correctly.

The issue is that as the months go on the new moon drifts all the way up to day 3 of a month (the other phases drift accordingly too) which breaks that alignment you are looking for. Again could play around with the Cycle Length or Cycle Adjust settings to try to get that closer but I don't think it can get to be exactly as you have listed right now. Having months with different days makes this hard.

I hope that gets you closer to your goal with the moons!
